backend/src/graphql/resolver/ContributionResolver.test.ts

```diff
@@ -160,11 +177,452 @@ describe('ContributionResolver', () => {
                   id: expect.any(Number),
                   amount: '100',
                   memo: 'Test env contribution',
+                  date: now,
+                  firstName: 'Bibi',
+                  lastName: 'Bloxberg',
+                  moderator: null,
+                  creation: ['1000', '1000', '900'],
+                  state: ContributionStatus.PENDING,
+                  messageCount: 0,
+                },
+              },
+            }),
+          )
+        })
+      })
+    })
+
+    describe('ClientRequestTime to 1st day of next month ahead server', () => {
+      beforeAll(async () => {
+        await cleanDB()
+        resetToken()
+
+        await userFactory(testEnv, bibiBloxberg)
+        // create one contribution with 100GDD for bibi one month ago in the past
+        await contributionFactory(testEnv, bibiBloxberg, capturedContribution100OneMonthAgo)
+        // set clientRequestTime at the 1st day of the next month against the backend time
+        const clientRequestTime = new Date()
+        clientRequestTime.setDate(1)
```

Comment:
solved
your example does not wrok because with setHours(4) your Date = 01.10.2022T04:00:00.000Z 
but with +09:00 you set the Timestring to 01.10.2022T04:00:00.000+09:00 which is in UTC 31.09.2022T19:00:00.000 and again the same month like the server